akka-actor/src/main/scala/akka/event/DeadLetterListener.scala

```diff
@@ -26,18 +30,52 @@ class DeadLetterListener extends Actor {
   override def postStop(): Unit =
     eventStream.unsubscribe(self)
 
-  def receive = {
-    case DeadLetter(message, snd, rcp) ⇒
-      count += 1
-      val origin = if (snd eq context.system.deadLetters) "without sender" else s"from $snd"
-      val done = maxCount != Int.MaxValue && count >= maxCount
-      val doneMsg = if (done) ", no more dead letters will be logged" else ""
-      eventStream.publish(Info(rcp.path.toString, rcp.getClass,
-        s"Message [${message.getClass.getName}] $origin to $rcp was not delivered. [$count] dead letters encountered$doneMsg. " +
-          s"If this is not an expected behavior, then [$rcp] may have terminated unexpectedly, " +
-          "This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' " +
-          "and 'akka.log-dead-letters-during-shutdown'."))
-      if (done) context.stop(self)
+  def receive: Receive = {
+
+    case DeadLetter(message, sender, receipt) ⇒
+      if (isAlwaysLoggingDeadLetters) {
+        logDeadLetter(message, sender, receipt, doneMsg = "")
+      } else {
+        suspendDuration match {
+          case duration: FiniteDuration ⇒
+            if (count == maxCount && suspendDeadline.isOverdue()) {
```

Comment:
Instead, you could keep on counting during the silence duration and log a summary afterwards about how many dead letters were skipped.